data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/RdsService.java

```diff
@@ -121,21 +133,23 @@ public void start(Buffer<Record<Event>> buffer) {
         }
 
         if (sourceConfig.isStreamEnabled()) {
-            BinlogClientFactory binaryLogClientFactory = new BinlogClientFactory(sourceConfig, rdsClient, dbMetadata);
+            ReplicationLogClientFactory replicationLogClientFactory = new ReplicationLogClientFactory(sourceConfig, rdsClient, dbMetadata);
```

Comment:
This class needs some refactor. We should follow single responsibility principle here. This class has dependency on MySQL and Postgres Client. We should move these dependencies to separate class.

Reply:
I did some refactoring and extracted out the creation of schemaManagers in separate classes. The `ReplicationLogClientFactory` already hides the creation of MySQL/Postgres clients.